bl/handler/tournament-handler.go

```diff
@@ -0,0 +1,109 @@
+package handler
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/kimbellG/kerror"
+	ttgrpc "github.com/kimbellG/tournament-bl/handler/grpc"
+	"github.com/kimbellG/tournament-bl/models"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+func (sh *ServiceHandler) CreateTournament(ctx context.Context, r *ttgrpc.CreateTournamentRequest) (*ttgrpc.CreateTournamentResponse, error) {
```

Comment:
Also I forgot to tell you that ctx must be passed all way down to your repository db function calls. All your queries to database must have ctx attached to them.